src/materialized/file_metadata.rs

```diff
@@ -126,79 +123,6 @@ impl TableProvider for FileMetadata {
     }
 }
 
-impl RowMetadataSource for FileMetadata {
```

Comment:
This implementation was moved to the `row_metadata` module and replaced with `ObjectStoreRowMetadataSource`, which allows mocking the file metadata if needed, which is useful in some test environments.